File: GameLogicLibrary/VBGameLogicLibrary/Coin.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace GameLogicLibrary
    
    Class Coin
        
        '.Publicly used object references
        Private r As Random = New Random(Environment.TickCount)
        
        '.Fields
        Private coin As Boolean
        
        '.Constructor that returns true or false. (Heads or Tails)
        Public Sub New()
            MyBase.New
            If (r.Next(2) = 0) Then
                Me.coin = true
            Else
                Me.coin = false
            End If
        End Sub
        
        '.Reset the coin field if another flip result is needed.
        Public Property Coins As Boolean
            Get
                Return Me.coin
            End Get
            Set
                If (r.Next(2) = 0) Then
                    Me.coin = true
                Else
                    Me.coin = false
                End If
            End Set
        End Property
    End Class
End Namespace

File: GameLogicLibrary/GameLogicLibrary/Coin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GameLogicLibrary {
	class Coin {
		//.Publicly used object references
		Random r = new Random(Environment.TickCount);
		
		//.Fields
		bool coin;

		//.Constructor that returns true or false. (Heads or Tails)
		public Coin() {
			if(r.Next(2) == 0)
				this.coin = true;
			else
				this.coin = false;
		}

		//.Reset the coin field if another flip result is needed.
		public bool Coins {
			get { return this.coin; }
			set {
				if(r.Next(2) == 0)
					this.coin = true;
				else
					this.coin = false;
			}
		}
	}
}
